Workiva/over_react_codemod
# CPLAT-15826 Codemod to add `ThemeProvider` to `react_dom.render`

## Motivation
  
In order to allow RMUI components to be used anywhere in Wdesk, we need to wrap all production and example `react_dom.render` calls in `ThemeProvider`.

This needs to be done before migrations take place. Otherwise, packages that perform their own react_dom.render calls could consume UI from other packages that have been migrated to RMUI without a ThemeProvider, resulting in the default MUI theming being displayed.

## Changes
  
- Add a `wk_theme_provider` codemod that wraps render calls in `ThemeProvider()..theme = wkTheme` and adds the `theme_provider.dart` import.
- Write tests.

#### Release Notes
  

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
        
      - [ ] CI passing and good test coverage
      - [ ] Pull down this branch and run: `pub global activate -s path .`
        - [ ] Run the codemod on another repo and verify that it works as expected:
          - `pub global run over_react_codemod:rmui_preparation --yes-to-all` 
          - Make sure `ThemeProvider`s are added inside `ErrorBoundary`s if they exist
          - Make sure render calls in test directories are not updated

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:


[contributing-review-types]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#manual-testing-criteria
